### ISSUE TITLE
NBA Lateswap: Unable to build lineup

### ISSUE THREAD

**Issue by holmes72** [DimaKudosh/pydfs-lineup-optimizer#339]
Why do I sometimes get LineupOptimizerException('Unable to build lineup') in link_players_with_positions when running NBA lateswap?  Does this happen when no lines can be optimized or is there some deeper meaning to this error?  How can I stop this from happening?

**Comment by holmes72** [comment]
I have a feeling this happens when a single lineup cannot be optimized and is already in its optimal state.  This isn't really an error, more of a condition.  However an error is raised, the process fails and no lines are optimized.

**Comment by holmes72** [comment]
I trapped this error and subbed the original line when this error was raised.  About half the lines that needed optimization were optimized, and the other half remained unchanged (even the "0" projection was used).  All rules were removed...just a vanilla late swap.

To duplicate this error, start with 30-or-so lines, wait until the last game time of the slate, change projections (drop a player to 0 and make sure there's a suitable replacement), then run late swap.  This should produce the "Unable to build lineup" error.

**Comment by tlabropoulos** [comment]
@holmes72 would you mind sharing how you got this resolved? Seem to be having lots of issues when arriving at the final game of the night.

**Comment by holmes72** [comment]
In lineup_optimizer.py, optimize_lineups, an unfeasible line throws an error and the program abends.  I removed throwing the error and instead I yield the original lineup.  Sometimes there are still issues with the line that require manual intervention (mostly in MLB...not so much with NBA), but at least this allows the other lines to optimize.  I added a ton of other stuff to handle this situation, but the core of the issue is the error being raised.
